src/formatter/text-format.tsx

```diff
@@ -34,12 +34,19 @@ export const TextFormat = ({ value, type, format, blankOnInvalid, locale }: ITex
     locale = TranslatorContext.context.locale;
     numeral.locale(locale);
   }
+
+  if (locale) {
```

Comment:
Instead of loading this every time the component is used may ne we can do this in this lib every time the locale changes. May be in setLocalr method in https://github.com/jhipster/react-jhipster/blob/main/src/language/translator-context.ts